### PR TITLE
Set policy CMP0167 to use FindBoost module

### LINKED ISSUE #1122: CMake policy CMP0167 (CMake 3.30 and FindBoost module)
According to the [policy page](https://cmake.org/cmake/help/latest/policy/CMP0167.html):
```
New in version 3.30.

The FindBoost module is removed.

CMake 3.29 and below provide a FindBoost module, but it needs constant updates
to keep up with upstream Boost releases. Upstream Boost 1.70 and above provide 
a BoostConfig.cmake package configuration file. find_package(Boost CONFIG)
finds the upstream package directly, without the find module.

CMake 3.30 and above prefer to not provide the FindBoost module so that
find_package(Boost) calls, without the CONFIG or NO_MODULE options, find the 
upstream BoostConfig.cmake directly. This policy provides compatibility for
projects that have not been ported to use the upstream Boost package.
```

I propose requiring at least Boost 1.70 (released 04/2019), and preferably Boost 1.75 (released 12/2020) because of CUDA. Our current Boost requirements are 1.56 in benchmarks/ and 1.67 in tests). And using `find_package(Boost CONFIG)` afterwards.

Alternatively, we may suppress the warning.

### PR DESCRIPTION
**Pull request by aprokop**
Fix #1122 for now.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,11 @@
 cmake_minimum_required(VERSION 3.16)
 project(ArborX CXX)
 
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.30.0)
+  message(STATUS "Setting policy CMP0167 to use FindBoost module")
+  cmake_policy(SET CMP0167 OLD)
+endif()
```
Comment (dalg24) [inline CMakeLists.txt:7]: Did you consider doing `find_package(Boost CONFIG)` instead?

Comment (aprokop) [reply CMakeLists.txt:7]: I did. That would not work with Boost 1.56-1.69.

**Comment by aprokop** [comment]
I tested locally with CMake 3.30, and it passed.

**Comment by aprokop** [comment]
@dalg24 Do you have any opposition to merging it in the current form, without worrying about updating Boost versions?

**Comment by dalg24** [comment]
> @dalg24 Do you have any opposition to merging it in the current form, without worrying about updating Boost versions?

Not blocking no, but also not seeing it as a priority.
The Boost dependency is not the same nature as the Kokkos one.  I feel like bumping the minimum version for Boost is the right choice as compare to what is proposed here.

**Comment by dalg24** [comment]
The CI reports a CMake warning.  Did you make sure it was ok to ignore?

**Pull request merged**